python/circuitsvis/attention.py

```diff
@@ -130,3 +143,468 @@ def attention_pattern(
         "AttentionPattern",
         **kwargs
     )
+
+
+
+
+help_strings_dicts = {
+    "large": "",
+    "small": r"""
+The vertical axis has query positions, the horizontal axis has key positions. You can hover over / click on the different heads to see attention probabilities per-head. You can also hover over the words printed at the bottom to see a different representation of the attention probabilities.<br><br>Note that this doesn't show you the actual probabilities - try using `mode = "lines"` or `mode = "large"` for that.<br><br>You can use the <b>Tokens</b> dropdown to switch from <b>Source 🡸 Destination</b> to <b>Destination 🡸 Source</b> (i.e. rather than seeing what a particular destination token attends to, you can see what attends to a particular source token).
+""",
+    "lines": r"""
+Hover over the text to see the attention paid to each token, and details of the computation.<br><br>The first two columns show the query and key vectors. The third columns shows their elementwise product. The fourth shows their dot product (which is the sum of their elementwise product, scaled by `sqrt(d_head)`). The fifth shows their probabilities.
+""",
+    "batch_dim": r"""
+<br><br>Your cache has a batch dimension, meaning you can select different sequences to show using the menu below.
+""",
+    "value-weighted": r"""
+<br><br>You've specified value-weighted attention, meaning every attention probability <code>A<sup>h</sup>[q, k]</code> will be replaced with <code>A<sup>h</sup>[q, k] * norm(v<sup>h</sup>[k]) / max<sub>k</sub>{norm(v<sup>h</sup>[k])}</code>. This more accurately reflects the size of the vector which is moved from source to destination. Note, this means they will sum to less than 1. Note that you can also choose <code>"info-weighted"</code> attention, which includes the <code>W<sup>h</sup><sub>O</sub></code> matrix.
+""",
+    "info-weighted": r"""
+<br><br>You've specified info-weighted attention, meaning every attention probability <code>A<sup>h</sup>[q, k]</code> will be replaced with <code>A<sup>h</sup>[q, k] * norm(v<sup>h</sup>[k] @ W<sup>h</sup><sub>O</sub>) / max<sub>k</sub>{norm(v<sup>h</sup>[k] @ W<sup>h</sup><sub>O</sub>)}</code>. This more accurately reflects the size of the vector which is moved from source to destination. Note, this means they will sum to less than 1.
+"""
+}
+
+
+
+
+def get_weighted_attention(
+    pattern: Float[Tensor, "*batch head seq_Q seq_K"],
+    model: HookedTransformer,
+    attention_type: Literal["standard", "value-weighted", "info-weighted"],
+    layers_and_heads: Optional[Union[int, List]] = None,
+    v: Optional[Float[Tensor, "*batch head seq_K d_head"]] = None,
+) -> Float[Tensor, "*batch head seq_Q seq_K"]:
+    '''
+    Returns attention probabilities (possibly value or info-weighted).
```

Comment:
Could you please add parameter information to the docstring?
- differences between value/info-weighted attention types
- v === value vector (?)
- layer_and_heads can be a layer index or a list of (layer, head) tuples